src/riscv.c

```diff
@@ -259,6 +262,62 @@ static void map_file(char **ram_loc, const char *name)
     exit(EXIT_FAILURE);
 }
 
+#define ALIGN_FDT(x) (((x) + (FDT_TAGSIZE) - 1) & ~((FDT_TAGSIZE) - 1))
+static char *realloc_property(char *fdt,
+                              int nodeoffset,
+                              const char *name,
+                              int newlen)
+{
+    /* FIXME: assume RAM is enough */
```

Comment:
I'm unsure if we need to consider the case of the preallocated memory being smaller than the required size to extend the DTB. Ideally, we can assume this won't happen.